src/Castle.Windsor.Tests/ReferenceTracker.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright 2018 Castle Project - http://www.castleproject.org/
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace CastleTests
+{
+	using System;
+
+	using NUnit.Framework;
+
+	public static class ReferenceTracker
+	{
+		/// <summary>
+		/// <para>
+		/// Use a lambda method to evaluate an expression returning an object instance to track.
+		/// </para>
+		/// <para>
+		/// (If the expression is evaluated directly inside the test method, the runtime will keep it alive
+		/// until the test method returns if the runtime is in debug mode. ReSharper’s test runner runs in
+		/// debug mode when the solution is configuration is Debug.)
+		/// </para>
+		/// </summary>
+		public static ReferenceTracker<T> Track<T>(Func<T> getInstanceToTrack) where T : class
+		{
+			return ReferenceTracker<T>.Track(getInstanceToTrack);
+		}
+	}
+
+	/// <summary>
+	/// A test helper encapsulating correct reference tracking.
+	/// </summary>
+	public struct ReferenceTracker<T> where T : class
```

Comment:
Would be `readonly struct` if you were interested in setting `<LangVersion>latest</LangVersion>` or `<LangVersion>7.2</LangVersion>` in `Directory.Build.props`.

Reply:
I am generally in favor of updating to the latest language version, but I happily defer that decision to @jonorossi.

One thing to look out for here is Mono compatibility. There has been a discussion around updating the Windsor build process and possibly set up a build on Mono (we already do this for Core). Mono now ships with the Roslyn compilers, but not always the latest ones. For example, at this time of writing, `<LangVersion>7.3</LangVersion>` wouldn't yet be supported, so updating the language version too far would be a roadblock for setting up a Mono build.

How about holding back this change for now, but open a new issue about converting *all* eligible structs to `readonly` once the project & build setup is definitely ready for it?

Reply:
Done, see https://github.com/castleproject/Windsor/issues/442.